# HG changeset patch
# User Rik <octave@nomad.inbox5.com>
# Date 1310846218 25200
#      Sat Jul 16 12:56:58 2011 -0700
# Node ID efd924e19ff71ec9510ba2d3422bb910b7e1bdbd
# Parent  66af8b914607a723fe96f09545ec18e10786999e
codesprint: Add tests for sparse_auto_mutate()

* ov-base.cc (sparse_auto_mutate): Add functional test.

diff --git a/src/ov-base.cc b/src/ov-base.cc
--- a/src/ov-base.cc
+++ b/src/ov-base.cc
@@ -1543,8 +1543,21 @@ s (1, :) = 1;\n\
 typeinfo (s)\n\
 @result{} matrix\n\
 @end group\n\
 @end example\n\
 @end deftypefn")
 {
   return SET_INTERNAL_VARIABLE (sparse_auto_mutate);
 }
+
+/*
+
+%!test
+ s = speye(3);
+ sparse_auto_mutate (false);
+ s(:, 1) = 1;
+ assert (typeinfo (s), "sparse matrix");
+ sparse_auto_mutate (true);
+ s(1, :) = 1;
+ assert (typeinfo (s), "matrix");
+
+*/
